backend/dataall/base/utils/naming_convention.py

```diff
@@ -56,6 +56,20 @@ def build_compliant_name(self) -> str:
         suffix = f'-{self.target_uri}' if len(self.target_uri) else ''
         return f"{slugify(self.resource_prefix + '-' + self.target_label[:(max_length - len(self.resource_prefix + self.target_uri))] + suffix, regex_pattern=fr'{regex}', separator=separator, lowercase=True)}"
 
+    def build_compliant_name_with_index(self, index: int = None) -> str:
+        """
+        Builds a compliant AWS resource name with an index at the end of the policy name
```

Comment:
Added comments to knw the developer / users to let them know how this functions is different from the other ( build_compliant_name ) and also added a note to let the user know why index string length is considered into generating the naming convention

Reply:
Understood!